metaflow/plugins/kfp/kfp.py

```diff
@@ -564,6 +565,49 @@ def _step_cli(self, node: DAGNode, task_id: str, user_code_retries: int) -> str:
         cmds.append(" ".join(entrypoint + top_level + step))
         return " && ".join(cmds)
 
+    @staticmethod
+    def _create_resource_based_node_type_toleration(
+        cpu: float, memory: float
+    ) -> Optional[V1Toleration]:
+        """Allow large enough pod to use higher cost nodes
+        TODO: Replace using AIP-5264 MutatingWebHook (or Validating) for AIP pod scheduling policies
+
+        The following node types are considered for setting the threshold:
+        c5.4xlarge: 16 vCPU, 32 GB
+        r5.12xlarge: 48 vCPU, 384 GB
+        Resource threshold are lower than machine resource boundary to take overheads into
+        account.
+
+        Toleration allows pods to utilize larger nodes without enforcement.
+        Setting a low threshold for using larger host allow more freedom for scheduler
+        and potentially higher utilization rate.
+
+        cpu: number of vCPU requested. Fractions are allowed.
+        memory: memory requested in GB (not GiB)
```

Comment:
Do you mind adding a short description on the cost issue that Rajesh mentioned (that is the reason for this PR)? e.g. something on the lines of `this toleration ensures small pods do not get scheduled on r5.12xlarge nodes and thus do not dramatically increase the cost of cluster usage`..